packages/interface/src/components/action-item/index.js

```diff
@@ -22,11 +23,44 @@ function ActionItemSlot( {
 			bubblesVirtually={ bubblesVirtually }
 			fillProps={ { as: Item, ...fillProps } }
 		>
-			{ ( fills ) =>
-				! isEmpty( fills ) && (
-					<Container { ...props }>{ fills }</Container>
-				)
-			}
+			{ ( fills ) => {
+				if ( isEmpty( Children.toArray( fills ) ) ) {
+					return null;
+				}
+
+				// Special handling exists for backward compatibility.
+				// It ensures that menu items created by plugin authors aren't
+				// duplicated with automatically injected menu items coming
+				// from pinnable plugin sidebars.
+				// @see https://github.com/WordPress/gutenberg/issues/14457
+				const initializedByPlugins = [];
+				Children.forEach(
+					fills,
+					( {
+						props: { __unstableInitSource, __unstableTarget },
+					} ) => {
+						if (
+							__unstableTarget &&
+							__unstableInitSource === 'plugin'
+						) {
+							initializedByPlugins.push( __unstableTarget );
+						}
+					}
+				);
+				const children = Children.map( fills, ( child ) => {
+					if (
+						child.props.__unstableInitSource === 'sidebar' &&
```

Comment:
It seems there's no need to check "sidebar" here right? Can't we just check !== "plugin"

Reply:
That would work as well, but the values can be either `sidebar` and `plugin` so both ways work.

Reply:
I was actually thinking we could probably make it a boolean: `__unstableExplicitMenuItem` or something

Reply:
I'll give it a try. Thank you for the good pointers 💯